unified-codes/tools/scripts/dgraph/migration/v2/import.js

```diff
@@ -0,0 +1,294 @@
+  const unspsc = row.getCell(31).value;
+  const nzulmUC2 = row.getCell(32).value;
+  const nzulmUC67 = row.getCell(33).value;
+  const snomed = row.getCell(34).value;
+  const drugbank = row.getCell(35).value;
+  const usfda = row.getCell(36).value;
+
+  const productDefinition = [
+    {
+      type: 'Product',
+      name: productName,
+      code: productCode,
+      properties: [
+        { type: 'code_nzulm', value: nzulmUC2 },
+        { type: 'code_rxnav', value: rxNavUC2 },
+        { type: 'code_unspsc', value: unspsc },
+        { type: 'code_who_atc', value: atcUC2 },
+        { type: 'ddd', value: dddUC2 },
+        { type: 'who_eml', value: emlUC2 },
+      ],
+    },
+    {
+      type: 'Route',
+      name: routeName,
+      code: routeCode,
+      properties: [],
+    },
+    {
+      type: 'DoseForm',
+      name: doseFormName,
+      code: doseFormCode,
+      properties: [],
+    },
+    {
+      type: 'DoseQualification',
+      name: doseQualificationName,
+      code: doseQualificationCode,
+      properties: [
+        { type: 'code_who_atc', value: atcUC5 },
+        { type: 'ddd', value: dddUC5 },
+      ],
+    },
+    {
+      type: 'DoseStrength',
+      name: doseStrengthName,
+      code: doseStrengthCode,
+      properties: [],
+    },
+    {
+      type: 'DoseUnit',
+      name: doseUnitName,
+      code: doseUnitCode,
+      properties: [],
+    },
+  ];
+
+  const [product] = productDefinition;
+
+  if (product.name.indexOf('/') > -1) {
+    combinationDrugs.push({
+      productName,
+      productCode,
+      combinations,
+    });
+  }
+
+  const synonymsArray = synonyms ? synonyms.split(',') : [];
+  await insertProduct(product, categoryCode, synonymsArray);
+
+  let parentIndex = 0;
+  let childIndex = 1;
+  while (childIndex < productDefinition.length) {
+    if (productDefinition[childIndex].name && productDefinition[childIndex].code) {
+      await insertChild(productDefinition[parentIndex], productDefinition[childIndex]);
+      parentIndex = childIndex;
+    }
+    childIndex++;
+  }
+};
+
+const insertProduct = async (product, categoryCode, synonymsArray) => {
+  // const propertiesQueries = propertyQueriesForNode(product);
+  // const propertiesMutations = propertyMutationsForNode(product);
+
+  const query = `query {
+    Category as var(func: eq(dgraph.type, Category)) @filter(eq(code, ${categoryCode}))
+    ${product.type} as var(func: eq(code, ${product.code})) 
+  }`;
+
+  const synonymMutation = synonymsArray.reduce(
+    (mutation, synonym, i) =>
+      `${mutation}uid(${product.type}) <name@alt${i}> "${synonym.trim()}" .\n`,
+    ''
+  );
+  const mutation = new dgraph.Mutation();
+  mutation.setSetNquads(`
+    uid(${product.type}) <name> "${product.name}" .
+    uid(${product.type}) <code> "${product.code}" .
+    uid(${product.type}) <dgraph.type> "${product.type}" .
+    uid(Category) <children> uid(${product.type}) .
+    ${synonymMutation}
+  `);
```

Comment:
Again, same thing, same thing (sorry 😢 ). Might be good to have queries as class methods/static fields, just for easier maintenance and stuff?